infer/environment.cc

```diff
@@ -23,6 +24,17 @@ core::TypePtr dropConstructor(core::Context ctx, core::Loc loc, core::TypePtr tp
     return tp;
 }
 
+bool typeTestReferencesVar(const InlinedVector<std::pair<cfg::LocalRef, core::TypePtr>, 1> &typeTest,
+                           cfg::LocalRef var) {
+    for (auto &test : typeTest) {
+        if (test.first == var) {
+            return true;
+        }
+    }
+    return false;
```

Comment:
What do you think about replacing this with a use of `absl::c_any_of` instead?

Reply:
does that work with pairs? I'm not looking for something that matches var exactly, I'm looking for a pair with var as the first item in the pair.

Reply:
I would think that you could just do something like the following, though you may have to make the type of `test` explicit:
```c++
return absl::c_any_of(typeTest, [var](auto &test) { return test.first == var; });
```

Reply:
Done!